qek/data/graphs.py

```diff
@@ -412,20 +454,31 @@ def __init__(
                 with `self.node_mapping`
              - `int` labels on edges, which may be converted into chemical
                 bounds with `self.edge_mapping`
-             - `int` attribute `y`, which will be used as a prediction target.
             The graph will not be changed.
         device: The device for which the sequence must be compiled.
         id: A unique identifier for the graph, used mostly for logging
             and displaying error messages.
     """
 
-    def ingest(self, graph: pyg_data.Data, device: pl.devices.Device, id: int) -> BaseGraph:
+    def ingest(
+        self, graph: Tuple[pyg_data.Data, int | None], device: pl.devices.Device, id: int
```

Comment:
```suggestion
        self, graph: tuple[pyg_data.Data, int | None], device: pl.devices.Device, id: int
```

Also, `id` is concretised here. Is it possible above ?

Reply:
Also, also: possible to concretise the return types for your `ingest` methods ?